tests/Analyze/Eval.hs

```diff
@@ -0,0 +1,126 @@
+import qualified Data.Default             as Default
+import qualified Data.HashMap.Strict      as HM
+import qualified Data.Map.Strict          as Map
+import           Data.SBV                 (literal, unliteral, writeArray)
+import qualified Data.SBV.Internals       as SBVI
+import qualified Data.Text                as T
+
+import           Pact.Analyze.Eval        (lasSucceeds, latticeState,
+                                           runAnalyze)
+import           Pact.Analyze.Eval.Term   (evalETerm)
+import           Pact.Analyze.Types       hiding (Object, Term)
+import           Pact.Analyze.Types.Eval  (aeDecimals, aeKeySets, mkAnalyzeEnv,
+                                           mkInitialAnalyzeState)
+import           Pact.Analyze.Util        (dummyInfo)
+
+import           Pact.Eval                (reduce)
+import           Pact.Repl                (initPureEvalEnv)
+import           Pact.Repl.Types          (LibState)
+import           Pact.Types.Runtime       (EvalEnv, PactError (..),
+                                           PactErrorType (EvalError), eeMsgBody,
+                                           runEval)
+import qualified Pact.Types.Term          as Pact
+
+import           Analyze.Gen
+
+
+data EvalResult
+  = EvalResult !(Pact.Term Pact.Ref)
+  | Discard
+  | EvalErr !String
+  | UnexpectedErr !String
+
+-- Evaluate a term via Pact
+pactEval
+  :: Pact.Term Pact.Ref
+  -> EvalEnv LibState
+  -> IO EvalResult
+pactEval pactTm evalEnv = (do
+    let evalState = Default.def
+    -- evaluate via pact, convert to analyze term
+    (pactVal, _) <- runEval evalState evalEnv (reduce pactTm)
+    Just pactVal' <- pure $ closed pactVal
+
+    pure $ EvalResult pactVal'
+  )
+    -- discard division by zero, on either the pact or analysis side
+    --
+    -- future work here is to make sure that if one side throws, the other
+    -- does as well.
+    `catch` (\(DivideByZero :: ArithException) -> pure Discard)
+    `catch` (\(pe@(PactError err _ _ msg) :: PactError) ->
+      case err of
+        EvalError ->
+          if "Division by 0" `T.isPrefixOf` msg ||
+             "Negative precision not allowed" `T.isPrefixOf` msg
+          then pure Discard
+          else pure $ EvalErr $ T.unpack msg
+        _ -> case msg of
+          "(enforce)"     -> pure $ EvalErr $ T.unpack msg
+          "(enforce-one)" -> pure $ EvalErr $ T.unpack msg
+          ""              -> pure $ UnexpectedErr $ show pe
+          _               -> pure $ UnexpectedErr $ T.unpack msg)
+
+-- Evaluate a term symbolically
+analyzeEval :: ETerm -> GenState -> IO (Either String ETerm)
+analyzeEval etm@(ESimple ty _tm) (GenState _ keysets decimals) = do
+  -- analyze setup
+  let tables = []
+      args = Map.empty
+      state0 = mkInitialAnalyzeState tables
+
+      tags = ModelTags Map.empty Map.empty Map.empty Map.empty Map.empty
+        -- this 'Located TVal' is never forced so we don't provide it
+        undefined
```

Comment:
should probably use `error` instead so we can track down problems if it ever ends up forced in the future